src/ripple/app/ledger/Ledger.cpp

```diff
@@ -958,142 +1018,109 @@ Ledger::read (Keylet const& k) const
 
 //------------------------------------------------------------------------------
 
-class Ledger::tx_iterator_impl
-    : public BasicView::iterator_impl
+auto
+Ledger::txsBegin() const ->
+    std::unique_ptr<txs_type::iter_base>
 {
-private:
-    SHAMap::iterator iter_;
-
-public:
-    explicit
-    tx_iterator_impl (SHAMap::iterator iter)
-        : iter_(iter)
-    {
-    }
-
-    std::unique_ptr<iterator_impl>
-    copy() const override
-    {
-        return std::make_unique<
-            tx_iterator_impl>(
-                iter_);
-    }
-
-    bool
-    equal (iterator_impl const& impl) const override
-    {
-        auto const& other = dynamic_cast<
-            tx_iterator_impl const&>(impl);
-        return iter_ == other.iter_;
-    }
-
-    void
-    increment() override
-    {
-        ++iter_;
-    }
+    return std::make_unique<
```

Comment:
These were the choices:
```
auto
Ledger::txsBegin() const ->
    std::unique_ptr<txs_type::iter_base>
{
    return std::make_unique<
        txs_iter_impl>(closed(),
            txMap_->begin(), *this);
}

auto
Ledger::txsBegin() const ->
    std::unique_ptr<txs_type::iter_base>
{
    return std::make_unique<txs_iter_impl>(
        closed(), txMap_->begin(), *this);
}
```

I went with the first one although, the second one is probably okay too.